lib/IR/IR.cpp

```diff
@@ -359,6 +359,24 @@ bool Instruction::isDataParallel() const {
   return false;
 }
 
+Instruction *Instruction::clone() const {
+  switch (getKind()) {
+  default:
+    llvm_unreachable("Unknown value kind");
+    break;
+#define DEF_INSTR(CLASS, NAME)                                                 \
+  case Kinded::Kind::CLASS##Kind: {                                            \
+    auto *X = llvm::cast<const CLASS>(this);                                   \
+    return X->clone();                                                         \
+    break;                                                                     \
+  }
+#define DEF_BACKEND_SPECIFIC_INSTR(CLASS, NAME) DEF_INSTR(CLASS, NAME)
+#define DEF_VALUE(CLASS, NAME)
+#include "glow/AutoGenInstr.def"
+  }
+  return nullptr;
```

Comment:
Huh, we need this? Assumed default case with llvm_unreachable was sufficient.

Reply:
Old habit! If a function has a return type different than void, it should return a value of this type ;-) Some (old) compilers always require that.